lib/matplotlib/collections.py

```diff
@@ -711,6 +711,7 @@ def set_alpha(self, alpha):
                 float(alpha)
             except TypeError:
                 raise TypeError('alpha must be a float or None')
+        self.update_dict['array'] = True
```

Comment:
Could you add an explanation here? I am still completely confused why this has to be here for setting alphas, but not anywhere else for any of the other colorsetting code.

Could this be the result of the collision of RGBA color specs and a separate alpha parameter?